src/Datadog.Trace/ExtensionMethods/SpanExtensions.cs

```diff
@@ -67,13 +67,31 @@ public static void AddTagsFromDbCommand(this Span span, IDbCommand command)
             }
         }
 
-        internal static void SetServerStatusCode(this Span span, int statusCode)
+        internal static void SetHttpServerStatusCode(this Span span, int statusCode)
         {
             string statusCodeString = HttpTags.ConvertStatusCodeToString(statusCode);
             span.SetTag(Tags.HttpStatusCode, statusCodeString);
 
-            // 5xx codes are server-side errors
-            if (statusCode / 100 == 5)
+            // Check the customers SERVER http statuses that should be marked as errors
+            if (Tracer.Instance.Settings.HttpServerErrorCodes.TryGetValue(statusCode, out _))
```

Comment:
This is a Dictionary lookup in a hot path. @kevingosse any idea here to reduce the overhead?

Reply:
Since http status codes range from 100 to 599, maybe a `BitArray` with size 600 and never use indexes 0-99? I doubt that doing the math to shift the indexes by 100 (0-499) on every request is worth the 12 bytes of memory that would save.

Reply:
Sounds good! worth checking the `BitArray` performance.

Reply:
So should I go with this approach instead of saving into a dictionary? Just wanted to confirm if it was good for everyone before proceeding.

Reply:
Actually, a simple `bool[600]` may be faster since it's a simple index lookup (`BitArray` has do [a few bitwise operations](https://github.com/dotnet/runtime/blob/7ae298e4f21a1e7015d0eb085ae7cbae2e9db31c/src/libraries/System.Collections/src/System/Collections/BitArray.cs#L319-L337)). It will be larger (~600 bytes vs ~75 bytes), but that seems like a fair trade-off.

Reply:
I would like to benchmark this, it's true the index operation is cheaper than bitwise operations, but 75 bytes have more probabilities to not be missed by a processor cache, than 600 bytes. so I really don't know who will perform faster depending of which scenario.

Reply:
stupids nanoseconds....

Reply:
Checking the ASM code, probably wins the index operation. the bitwise has double bound check before doing the operations:

```
C.Get(Int32)
    L0000: sub rsp, 0x28
    L0004: cmp edx, [rcx+0x18]
    L0007: jae short L002c

    L0009: mov rax, [rcx+8]
    L000d: mov ecx, edx
    L000f: sar ecx, 5
    L0012: cmp ecx, [rax+8]
    L0015: jae short L0034


    L0017: movsxd rcx, ecx
    L001a: mov eax, [rax+rcx*4+0x10]
    L001e: bt eax, edx
    L0021: setb al
    L0024: movzx eax, al
    L0027: add rsp, 0x28
    L002b: ret


    L002c: mov ecx, edx
    L002e: call C.ThrowArgumentOutOfRangeException(Int32)
    L0033: int3
    L0034: call 0x00007ffae0a0fc60
    L0039: int3
```

Meanwhile accessing the index directly has just one bound check:

```
C.GetArray(Int32)
    L0000: sub rsp, 0x28
    L0004: mov rax, [rcx+0x10]
    L0008: cmp edx, [rax+8]
    L000b: jae short L001a


    L000d: movsxd rdx, edx
    L0010: movzx eax, byte ptr [rax+rdx+0x10]
    L0015: add rsp, 0x28
    L0019: ret


    L001a: call 0x00007ffae0a0fc60
    L001f: int3
```

There is a way to remove one bound check to the bitwise, but it's totally outside of the scope of this PR.

To unblock you, go with the Array[600] method. We will revisit this if we need to squish those ns (probably not).

Reply:
Hahahaha sounds good, thank you for the detailed comparison!